scripts/data/climate/colorize.py

```diff
@@ -0,0 +1,205 @@
+                   (246, 155, 58),
+                   (62, 178, 189),
+                   (97, 212, 231),
+                   (3, 116, 116),
+                   (0, 49, 96)],
+        'position': [0.0, 0.08, 0.16, 0.24, 0.32, 1.0],
+    },
+    'evi': {
+        'colors': [(229, 229, 229), (182, 165, 134), (160, 138, 91),
+                   (138, 111, 49), (140, 127, 43), (142, 143, 37),
+                   (144, 159, 31), (146, 175, 25), (138, 177, 21),
+                   (119, 165, 18), (99, 154, 15), (80, 142, 12),
+                   (60, 131, 9), (41, 119, 6), (22, 108, 3),
+                   (3, 97, 0), (0, 23, 0)],
+        'position': [0, 0.04, 0.08, 0.12, 0.16, 0.20, 0.24, 0.28, 0.32, 0.36,
+                     0.40, 0.44, 0.48, 0.52, 0.56, 0.60, 1],
+    }
+}
+
+
+def make_cmap(colors, position=None, bit=False):
+    """
+    ** This code provided as is from:
+    https://github.com/WikiWatershed/model-my-watershed/issues/2122
+    **
+
+    make_cmap takes a list of tuples which contain RGB values. The RGB
+    values may either be in 8-bit [0 to 255] (in which bit must be set to
+    True when called) or arithmetic [0 to 1] (default). make_cmap returns
+    a cmap with equally spaced colors.
+    Arrange your tuples so that the first color is the lowest value for the
+    colorbar and the last is the highest.
+    position contains values from 0 to 1 to dictate the location of each color.
+    """
+    bit_rgb = np.linspace(0, 1, 256)
+    if position is None:
+        position = np.linspace(0, 1, len(colors))
+    else:
+        if len(position) != len(colors):
+            sys.exit("position length must be the same as colors")
+        elif position[0] != 0 or position[-1] != 1:
+            sys.exit("position must start with 0 and end with 1")
+    if bit:
+        for i in range(len(colors)):
+            colors[i] = (bit_rgb[colors[i][0]],
+                         bit_rgb[colors[i][1]],
+                         bit_rgb[colors[i][2]])
+    cdict = {'red': [], 'green': [], 'blue': []}
+    for pos, color in zip(position, colors):
+        cdict['red'].append((pos, color[0], color[0]))
+        cdict['green'].append((pos, color[1], color[1]))
+        cdict['blue'].append((pos, color[2], color[2]))
+
+    cmap = mpl.colors.LinearSegmentedColormap('my_colormap', cdict, 256)
+    return cmap
+
+
+def normalize_to_rgba(var, arr, colormap):
+    """
+    ** This code was lightly edited for clarity and correnctness from:
+    https://github.com/WikiWatershed/model-my-watershed/issues/2122
+    **
+
+    Normalize raster values around display values ranges and convert
+    to segmented linear interpolated rgba values for rendering
+    """
+
+    values = {
+        'tmean': {
+            'min': 0,
+            'max': 60
+        },
+        'ppt': {
+            'min': 0,
+            'max': 500
+        },
+        'evi': {
+            'min': 0,
+            'max': 10000
+        }
+    }
+
+    if var == 'tmean':
```

Comment:
I've tweaked this a little bit. Please see https://github.com/WikiWatershed/model-my-watershed/issues/2122#issuecomment-324168504